assets/js/plugin-check-admin.js

```diff
@@ -172,14 +172,41 @@
 	 * @param {Object} data The response data.
 	 */
 	async function runChecks( data ) {
+		let isSuccessMessage = true;
+		let messageClass = '';
+		let messageText = '';
 		for ( let i = 0; i < data.checks.length; i++ ) {
 			try {
 				const results = await runCheck( data.plugin, data.checks[ i ] );
+				const errorsLength = Object.values( results.errors ).length;
+				const warningsLength = Object.values( results.warnings ).length;
+				if (
+					isSuccessMessage &&
+					( errorsLength > 0 || warningsLength > 0 )
+				) {
+					isSuccessMessage = false;
+				}
 				renderResults( results );
 			} catch ( e ) {
 				// Ignore for now.
 			}
 		}
+
+		if ( isSuccessMessage ) {
+			messageClass = 'notice-success';
+			messageText = pluginCheck.successMessage;
+		} else {
+			messageClass = 'notice-error';
+			messageText = pluginCheck.errorMessage;
+		}
+
+		resultsContainer.innerHTML += renderTemplate(
+			'plugin-check-results-complete',
+			{
+				class: messageClass,
+				message: messageText,
+			}
+		);
```

Comment:
Can you please move all the new logic back into the `renderResults()` function below? That's where this template was originally rendered, so I'm not sure why it would need to be moved here now, which intertwines it more with other higher-level logic.

All new logic is based on the `results` variable, which is passed to `renderResults()` anyway, so we can have the logic in there without any notable adjustments.

Reply:
@felixarntz @mukeshpanchal27 this came from my previous review https://github.com/10up/plugin-check/pull/160#discussion_r1190137462

We call `renderResults` for each check, so if we keep this logic here the final 'Checks Complete' message will display multiple times. Moving this here means the message would only display once after all checks have been run.

Reply:
@mukeshpanchal27 Ahh, I see, thanks for clarifying. In that case, I'd suggest we create a new function, e.g. `renderResultsMessage()`, with the new logic instead of adding that to the other higher-level function, which makes it more complex.

Reply:
Thanks @felixarntz for the review. Addressed in https://github.com/10up/plugin-check/pull/160/commits/61dc1cc450c5619cf0a88cb42abd7199bc226265